git-fixup

```diff
@@ -47,9 +47,16 @@ function print_sha () {
     git --no-pager log --format="%H [$type] %s <%ae>" -n 1 "$sha"
 }
 
+op="fixup"
+case "$1" in
+    -s|--squash)
+        op="squash"
+        shift
+        ;;
+esac
+
 if test $# -eq 1; then
-    git rev-parse --verify "$1" > /dev/null || exit 1
-    git commit -m "fixup! $1";
+    git commit --$op=$1
     exit 0
```

Comment:
should exit with error if the commit fails. Could change it to a plain exit without arguments to reuse that exit code of git commit